stubs/Reflection.phpstub

```diff
@@ -92,32 +471,74 @@ class ReflectionProperty implements Reflector
     /**
      * @since 7.4
      * @psalm-assert-if-true ReflectionType $this->getType()
+     * @psalm-pure
      */
     public function hasType() : bool {}
 
     /**
      * @since 7.4
-     * @psalm-mutation-free
+     * @psalm-pure
      */
     public function getType() : ?ReflectionType {}
 
+    /** @psalm-pure */
+    public function isPublic(): bool {}
+
+    /** @psalm-pure */
+    public function isPrivate(): bool {}
+
+    /** @psalm-pure */
+    public function isProtected(): bool {}
+
+    /** @psalm-pure */
+    public function isStatic(): bool {}
+
+    /** @psalm-pure */
+    public function isDefault(): bool {}
+
     /**
-    * @since 8.0
-    */
+     * @return int-mask-of<self::IS_>
+     * @psalm-pure
+     */
+    public function getModifiers(): int {}
+
+    /** @psalm-pure */
+    public function getDeclaringClass(): ReflectionClass {}
+
+    /**
+     * @return non-empty-string|false
+     *
+     * @psalm-pure
+     */
+    public function getDocComment() {}
+
+    /**
+     * @since 8.0
+     * @psalm-pure
+     */
     public function hasDefaultValue(): bool {}
 
     /**
-    * @since 8.0
-    */
+     * @return mixed
+     *
+     * @psalm-pure
+     */
+    public function getDefaultValue() {}
+
+    /**
+     * @since 8.0
+     * @psalm-pure
+     */
     public function isPromoted(): bool {}
 
     /**
-    * @since 8.1
-    */
+     * @since 8.1
+     * @psalm-pure
+     */
     public function isReadOnly(): bool {}
 }
 
-class ReflectionMethod implements Reflector
+class ReflectionMethod extends ReflectionFunctionAbstract
```

Comment:
Possibly some trouble with this? This is the analysis of PHPUnit after my changes:

```
ERROR: MissingThrowsDocblock - src/Framework/Constraint/ObjectEquals.php:64:28 - ReflectionException is thrown but not caught - please either catch or add a @throws annotation (see https://psalm.dev/169)
        $method = $object->getMethod($this->method);

ERROR: ArgumentTypeCoercion - src/Framework/Constraint/ObjectEquals.php:64:38 - Argument 1 of ReflectionObject::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
        $method = $object->getMethod($this->method);

ERROR: MissingThrowsDocblock - src/Framework/MockObject/Generator.php:741:40 - ReflectionException is thrown but not caught - please either catch or add a @throws annotation (see https://psalm.dev/169)
                $cloneMethod = $class->getMethod('__clone');

ERROR: ArgumentTypeCoercion - src/Framework/MockObject/Generator.php:770:49 - Argument 1 of ReflectionClass::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
                    $method = $class->getMethod($methodName);

ERROR: MissingThrowsDocblock - src/Framework/TestCase.php:1839:21 - ReflectionException is thrown but not caught - please either catch or add a @throws annotation (see https://psalm.dev/169)
            $class->getMethod($methodName)

ERROR: ArgumentTypeCoercion - src/Framework/TestCase.php:1839:31 - Argument 1 of ReflectionClass::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
            $class->getMethod($methodName)

ERROR: ArgumentTypeCoercion - src/Framework/TestCase.php:1992:38 - Argument 1 of ReflectionObject::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
               $reflector->getMethod($methodName)->getDeclaringClass()->getName() === self::class;

ERROR: ArgumentTypeCoercion - src/Framework/TestSuite.php:615:38 - Argument 1 of ReflectionClass::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
               $reflector->getMethod($methodName)->getDeclaringClass()->getName() === TestCase::class;

ERROR: ArgumentTypeCoercion - src/Metadata/Api/DataProvider.php:96:45 - Argument 1 of ReflectionClass::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
                $method = $class->getMethod($_dataProvider->methodName());

ERROR: UndefinedConstant - src/Util/Reflection.php:27:45 - Constant ReflectionMethod::IS_PUBLIC is not defined (see https://psalm.dev/020)
        return $this->filterMethods($class, ReflectionMethod::IS_PUBLIC);
```

Most notably: 

```
ERROR: UndefinedConstant - src/Util/Reflection.php:27:45 - Constant ReflectionMethod::IS_PUBLIC is not defined (see https://psalm.dev/020)
        return $this->filterMethods($class, ReflectionMethod::IS_PUBLIC);
```

Unsure about this one - this has to do with the stub loading mechanism, I think :thinking:

Reply:
Same in my local test, so more work needed here :cold_sweat: 

```
ERROR: UndefinedConstant - reflection-test.php:41:53 - Constant ReflectionMethod::IS_PUBLIC is not defined (see https://psalm.dev/020)
$results = [$name, $unionTypes, $intersectionTypes, \ReflectionMethod::IS_PUBLIC];
```

I wonder if constants in declared reflection symbols are not merged with stubs :thinking:

Reply:
As per 68d88c546b286a5fbdabe4728f6be9dd1db0dbe0, I migrated everything to `visitStubFiles`, which fixed most issues :)

Reply:
PHPUnit real-world test looks much better now:

```
ERROR: MissingThrowsDocblock - src/Framework/Constraint/ObjectEquals.php:64:28 - ReflectionException is thrown but not caught - please either catch or add a @throws annotation (see https://psalm.dev/169)
        $method = $object->getMethod($this->method);

ERROR: ArgumentTypeCoercion - src/Framework/Constraint/ObjectEquals.php:64:38 - Argument 1 of ReflectionObject::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
        $method = $object->getMethod($this->method);

ERROR: MissingThrowsDocblock - src/Framework/MockObject/Generator.php:741:40 - ReflectionException is thrown but not caught - please either catch or add a @throws annotation (see https://psalm.dev/169)
                $cloneMethod = $class->getMethod('__clone');

ERROR: ArgumentTypeCoercion - src/Framework/MockObject/Generator.php:770:49 - Argument 1 of ReflectionClass::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
                    $method = $class->getMethod($methodName);

ERROR: MissingThrowsDocblock - src/Framework/TestCase.php:1839:21 - ReflectionException is thrown but not caught - please either catch or add a @throws annotation (see https://psalm.dev/169)
            $class->getMethod($methodName)

ERROR: ArgumentTypeCoercion - src/Framework/TestCase.php:1839:31 - Argument 1 of ReflectionClass::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
            $class->getMethod($methodName)

ERROR: ArgumentTypeCoercion - src/Framework/TestCase.php:1992:38 - Argument 1 of ReflectionObject::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
               $reflector->getMethod($methodName)->getDeclaringClass()->getName() === self::class;

ERROR: ArgumentTypeCoercion - src/Framework/TestSuite.php:615:38 - Argument 1 of ReflectionClass::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
               $reflector->getMethod($methodName)->getDeclaringClass()->getName() === TestCase::class;

ERROR: ArgumentTypeCoercion - src/Metadata/Api/DataProvider.php:96:45 - Argument 1 of ReflectionClass::getMethod expects non-empty-string, but parent type string provided (see https://psalm.dev/193)
                $method = $class->getMethod($_dataProvider->methodName());

ERROR: ArgumentTypeCoercion - src/Util/Reflection.php:45:37 - Argument 1 of ReflectionClass::getMethods expects 0|1|2|3|4|5|6|7|16|17|18|19|20|21|22|23|32|33|34|35|36|37|38|39|48|49|50|51|52|53|54|55|64|65|66|67|68|69|70|71|80|81|82|83|84|85|86|87|96|97|98|99|100|101|102|103|112|113|114|115|116|117|118|119|null, but parent type int|null provided (see https://psalm.dev/193)
        foreach ($class->getMethods($filter) as $method) {

```